### DISCUSSION TITLE
标准 K8s 集群离线安装 - DaoCloud Enterprise

### DISCUSSION THREAD

**Discussion by giscus** [DaoCloud/DaoCloud-docs#1041] [Announcements]
# 标准 K8s 集群离线安装 - DaoCloud Enterprise

DaoCloud Enterprise 5.0 下一代容器化平台集大成者，引领云原生浪潮，推动数字化转型。

https://docs.daocloud.io/install/community/k8s/offline/

**Comment by kodyssss** [comment]
下载和安装-2小点-看不懂脚本的小白请按照我的方式进行部署，可以顺利安装。
1.添加环境变量
export  REGISTRY_ADDR=<你自己的harbor仓库地址>
export  REGISTRY_USER=<harbor的登录用户名>
export  REGISTRY_PASS=<harbor的登录密码>
2.在harbor中创建仓库
创建的仓库名称为 docker.m.daocloud.io；quay.m.daocloud.io；elastic.m.daocloud.io；ghcr.io；quay.io；release.daocloud.io；
3.在执行
./offline_image_handler.sh import

**Reply by kodyssss** [reply]
# 指定镜像仓库地址
export REGISTRY_ADDR=registry.daocloud.io:30080 
这个地址为你自己的 harbor的地址！！！！！！
另外：.skopeo没法创建仓库，哪怕用的管理员权限!!!！！
./offline_image_handler.sh这个脚本的逻辑就是读取offline下的image.list然后读取第一列和第二列再执行 skopeo copy;但因为skopeo工具没办法自动创建仓库，
会报错：“FATA[0000] trying to reuse blob sha256:da62b97c2205cc3f43f9f9cb706083bc67f0c3a9ad6f549d926dda4bb7e6c5ee at destination: checking whether a blob sha256:da62b97c2205cc3f43f9f9cb706083bc67f0c3a9ad6f549d926dda4bb7e6c5ee exists in harbor.com:543/docker.m.daocloud.io/grafana/grafana: authentication required”
需要创建“docker.m.daocloud.io；quay.m.daocloud.io；elastic.m.daocloud.io；ghcr.io；quay.io；release.daocloud.io；gcr.m.daocloud.io；ghcr.m.daocloud.io；k8s-gcr.m.daocloud.io” 这些所有仓库再执行./offline_image_handler.sh import.<默认执行也是import>
此时查看仓库就有镜像了。
